File: src/main/java/com/ascargon/rocketshow/lighting/LightingUniverse.java
package com.ascargon.rocketshow.lighting;

import jakarta.xml.bind.annotation.XmlRootElement;
import lombok.Getter;
import lombok.Setter;

import java.util.UUID;

@XmlRootElement
@Getter
@Setter
public class LightingUniverse {

    private String uuid = String.valueOf(UUID.randomUUID());
    private String name;
    private Integer olaUniverseId;
    private String olaOutputPortId;
    // OLA device name (e.g. "E1.31 (DMX over ACN) [192.168.1.200]") used to re-match the port ID after OLA restarts
    private String olaOutputPortDevice;

}

File: src/main/webapp/src/app/models/lighting-universe.ts
export class LightingUniverse {
  uuid: string;
  name: string = "";
  olaUniverseId: number = 1;
  olaOutputPortId: string = "";
  olaOutputPortDevice: string = "";

  constructor(data?: any) {
    if (!data) {
      return;
    }

    this.uuid = data.uuid;
    this.name = data.name;
    this.olaUniverseId = data.olaUniverseId;
    this.olaOutputPortId = data.olaOutputPortId || "";
    this.olaOutputPortDevice = data.olaOutputPortDevice || "";
  }
}
